sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/io/Failure.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.io;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.annotations.Internal;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+
+/** A generic failure of an SQL transform. */
+@Internal
+@Experimental(Kind.SCHEMAS)
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class Failure {
+  /** A JSON representation of the payload which has failed. */
+  public abstract String getPayload();
+  /** Information about the cause of the failure. */
+  public abstract String getError();
+
+  public static Builder newBuilder() {
+    return new AutoValue_Failure.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setPayload(String payload);
+
+    public abstract Builder setError(String error);
```

Comment:
For payload, will be better to be a ByteString type?

Reply:
Similarly for error?

Reply:
For error: No, I think String is better since this is intended to be a human-readable error.
For payload: Yes, this should probably be byte[]. It can't be ByteString I don't think since core cannot depend on protobuf? Changed.

Reply:
byte[] is ok. Good point on the protobuf dependency.